[thrift] Build is failing on 0.22.0

### Operating system

Linux RHEL 9.6

### Compiler

gcc (GCC) 11.5.0 20240719 (Red Hat 11.5.0-5)

### Steps to reproduce the behavior

```Shell
Create a vcpkg project that depends on thrift only,
Have a CMakeLists.txt with:

project(thrift_issue)
find_package(Thrift CONFIG REQUIRED)
find_program(THRIFT_EXECUTABLE thrift)
```

### Failure logs

CMake Error at vcpkg_installed/evs-blue-x64-linux-dynamic/share/thrift/ThriftConfig.cmake:32 (message):
  File or directory
  /home/dev/thrift_issue/vcpkg_installed/evs-blue-x64-linux-dynamic/bin/../tools/thrift
  referenced by variable THRIFT_BIN_DIR does not exist !
Call Stack (most recent call first):
  vcpkg_installed/evs-blue-x64-linux-dynamic/share/thrift/ThriftConfig.cmake:53 (set_and_check)
  /home/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:898 (_find_package)
  CMakeLists.txt:7 (find_package)


### Additional context

This is coming from here: https://github.com/microsoft/vcpkg/pull/46278/commits/6006cc465bd4c925625c0beea15130fec1574685#diff-14cf34fa8e1d648d6b4036d240586ce8e86f7a4c33fc7645a2b62a1cd45eafecR12
Actually in my case, there is no bin folder, so one cannot use bin/../tools

--- a/ports/thrift/fix_missing_quotes_in_config_and_bin_path.patch
+++ b/ports/thrift/fix_missing_quotes_in_config_and_bin_path.patch
@@ -2,14 +2,15 @@ diff --git a/build/cmake/ThriftConfig.cmake.in b/build/cmake/ThriftConfig.cmake.
 index f52480104..616dbeda6 100644
 --- a/build/cmake/ThriftConfig.cmake.in
 +++ b/build/cmake/ThriftConfig.cmake.in
-@@ -25,8 +25,8 @@ set_and_check(THRIFT_CMAKE_DIR "${CMAKE_CURRENT_LIST_DIR}")
+@@ -25,8 +25,9 @@ set_and_check(THRIFT_CMAKE_DIR "${CMAKE_CURRENT_LIST_DIR}")
  
  set_and_check(THRIFT_INCLUDE_DIR "@PACKAGE_INCLUDE_INSTALL_DIR@/thrift")
  
 -if(@BUILD_COMPILER@)
 -    set_and_check(THRIFT_BIN_DIR "@PACKAGE_BIN_INSTALL_DIR@")
 +if("@BUILD_COMPILER@")
-+    set_and_check(THRIFT_BIN_DIR "@PACKAGE_BIN_INSTALL_DIR@/../tools/thrift")
++    get_filename_component(PARENT_DIR "@PACKAGE_BIN_INSTALL_DIR@" DIRECTORY)
++    set_and_check(THRIFT_BIN_DIR "${PARENT_DIR}/tools/thrift")
      if(NOT DEFINED THRIFT_COMPILER)
          set(THRIFT_COMPILER "${THRIFT_BIN_DIR}/thrift@CMAKE_EXECUTABLE_SUFFIX@")
      endif()

--- a/ports/thrift/vcpkg.json
+++ b/ports/thrift/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "thrift",
   "version": "0.22.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Apache Thrift is a software project spanning a variety of programming languages and use cases. Our goal is to make reliable, performant communication and data serialization across languages as efficient and seamless as possible.",
   "homepage": "https://github.com/apache/thrift",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -9926,7 +9926,7 @@
     },
     "thrift": {
       "baseline": "0.22.0",
-      "port-version": 1
+      "port-version": 2
     },
     "tidy-html5": {
       "baseline": "5.8.0",

--- a/versions/t-/thrift.json
+++ b/versions/t-/thrift.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "2c69712275b12812b2464254884257cd47673d2c",
+      "version": "0.22.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "37ecae27a996313cbd6862b3c33aa026eda52692",
       "version": "0.22.0",
       "port-version": 1

